scapy/asn1fields.py

```diff
@@ -70,13 +69,18 @@ def m2i(self, pkt, s):
 
         Regarding other fields, we might need to know whether encoding went
         as expected or not. Noticeably, input methods from cert.py expect
-        certain exceptions to be raised. Hence default flexible_tag is False,
-        but we provide a FLEXIBLE_TAGS switch for debugging purposes.
+        certain exceptions to be raised. Hence default flexible_tag is False.
         """
-        s = BER_tagging_dec(s, hidden_tag=self.ASN1_tag,
-                            implicit_tag=self.implicit_tag,
-                            explicit_tag=self.explicit_tag,
-                            safe=self.flexible_tag)
+        diff_tag, s = BER_tagging_dec(s, hidden_tag=self.ASN1_tag,
+                                      implicit_tag=self.implicit_tag,
+                                      explicit_tag=self.explicit_tag,
+                                      safe=self.flexible_tag)
+        if diff_tag is not None:
+            # cannot happen without flexible_tag being True
```

Comment:
Perhaphs, checking for "flexible_tag == True" and raising an exception otherwise will be safer. This  applies to the other occurrences of this Python comment.